main/fastcgi.c

```diff
@@ -1469,6 +1474,7 @@ int fcgi_accept_request(fcgi_request *req)
 			return -1;
 		}
 		if (fcgi_read_request(req)) {
+			req->hook.on_read();
```

Comment:
Ahhh, I get it know. The poll/select is there to add a 5 second timeout for the initial read after accept. Of course this should not be there for a keepalive connection.